src/components/organisms/AdminVenueView/components/EventsView/CsvImportModal.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { useCallback, useState } from "react";
+import { Modal } from "react-bootstrap";
+import { useAsyncFn } from "react-use";
+import classNames from "classnames";
+import { ParseError, ParseMeta } from "papaparse";
+
+import { importEventsBatch, ImportEventsBatchResult } from "api/events";
+
+import { parseCsv, ParseData, transformImportToEvents } from "utils/csv";
+
+import { ButtonNG } from "components/atoms/ButtonNG/ButtonNG";
+import {
+  FileButton,
+  FileButtonOnChangeData,
+} from "components/atoms/FileButton/FileButton";
+
+import "./CsvImportModal.scss";
+
+export interface CsvImportModalProps {
+  onHide: () => void;
+  show: boolean;
+}
+
+export const CsvImportModal: React.FC<CsvImportModalProps> = ({
+  onHide,
+  show,
+}) => {
+  const [parsedData, setParsedData] = useState<ParseData[] | undefined>();
+  const [parsedMeta, setParsedMeta] = useState<ParseMeta | undefined>();
+  const [parsedErrors, setParsedErrors] = useState<ParseError[] | undefined>();
+  const [isParsing, setIsParsing] = useState<boolean>(false);
+  const [importResult, setImportResult] = useState<ImportEventsBatchResult>();
+
+  const clearParsed = useCallback(() => {
+    setParsedData(undefined);
+    setParsedMeta(undefined);
+    setParsedErrors(undefined);
+    setImportResult(undefined);
+  }, []);
+
+  const downloadExample = useCallback(() => {
+    const link = document.createElement("a");
```

Comment:
This is wrong on many levels, can we use something else?

Reply:
Too generic. Why is it wrong per each level?

Reply:
> Because you are creating an html element and manipulating the DOM tree. This is not react-friendly and is a very bad practice.

This was copied from how the schedule download works. If it wasn't React friendly for it as well, what is the friendly way?

Reply:
Can't you just send a request to the url and download the file?

Reply:
I can't prove a negative. If you think it is possible, please provide example. FYI here is the compatibility page for the "send a request" API https://caniuse.com/?search=downloads

Reply:
Here is a "2019 modern" version of it: https://stackoverflow.com/a/9834261

Reply:
Here is a version like what is already used in the project to download a schedule https://stackoverflow.com/a/23013574